LabExT/Measurements/InsertionLossSweep.py

```diff
@@ -181,7 +207,61 @@ def algorithm(self, device, data, instruments, parameters):
         self.instr_laser.close()
         self.instr_pm.close()
 
+        # apply reference to data
+        if parameters['file path to reference meas.'].value.strip():
+            self.apply_reference_to_data(data=data)
+        else:
+            data['values']['referenced transmission [dB]'] = []
+
+        # if user wants to save disk space by discarding raw data, do it now
+        if parameters['discard raw transmission data'].value:
+            data['values']['transmission [dBm]'] = []
+
         # sanity check if data contains all necessary keys
         self._check_data(data)
 
         return data
+
+    def check_if_reference_valid(self, parameters):
+
+        # load reference file
+        ref_fp = parameters['file path to reference meas.'].value.strip()
+        try:
+            with open(ref_fp, 'r') as fp:
+                ref_raw_data = json.load(fp=fp)
+        except FileNotFoundError:
+            raise FileNotFoundError(f'Reference file "{ref_fp:s}" not found.')
+        except json.decoder.JSONDecodeError:
+            raise RuntimeError(f'Reference file "{ref_fp:s}" could not be JSON decoded.')
+
+        # check reference parameters
+        check_params = ['wavelength start', 'wavelength stop', 'wavelength step', 'sweep speed', 'laser power']
+        for pname in check_params:
+            if ref_raw_data['measurement settings'][pname]['value'] != parameters[pname].value:
+                raise ValueError(f'Parameters used to record reference in file "{ref_fp:s}" do not match this '
+                                 f"ToDo's settings. Make sure you are using the same settings in the ToDo as "
+                                 f"were used to record the reference file for these parameters: {check_params}.")
+
+        self.ref_data = {
+            'wl': np.array(ref_raw_data['values']['wavelength [nm]']),
+            'tm': np.array(ref_raw_data['values']['transmission [dBm]'])
+        }
+
+    def apply_reference_to_data(self, data):
+
+        assert self.ref_data is not None, 'Reference data is not loaded!'
+
+        # get data back to numpy arrays
+        rec_wl = np.array(data['values']['wavelength [nm]'])
+        rec_tm = np.array(data['values']['transmission [dBm]'])
+
+        # notify user if wavelengths differ > 1pm
+        if any(np.abs(rec_wl - self.ref_data['wl']) > 1e-3):
+            self.logger.warning('Referenced and recorded wavelengths differ by > 1pm!')
```

Comment:
If one uses a smaller wavelength span than the calibration, this will raise an Exception. This came to mind because our maximal power of our lasers drops the further away from 1550 nm we are and someone might want to measure are higher power. Then he would need to perform a new calibration.